pygmt/src/grdhisteq.py

```diff
@@ -13,6 +12,7 @@
     fmt_docstring,
     kwargs_to_strings,
     use_alias,
+    validate_output_type,
```

Comment:
Just on the naming, should we specify that this applies to tabular outputs? E.g. use something like `validate_output_table_type`? There are some functions like `pygmt.grdhisteq.equalize_grid` that has code looking like this:

```python
with GMTTempFile(suffix=".nc") as tmpfile:
    if isinstance(outgrid, str):
        output_type = "file"
    elif outgrid is None:
        output_type = "xarray"
        outgrid = tmpfile.name
    else:
        raise GMTInvalidInput("Must specify 'outgrid' as a string or None.")
```

If we ever decide to have more output_types for grids (e.g. NetCDF or TIFF or xarray or ???), then there might be a `validate_output_grid_type`.

Reply:
I was thinking about adding a new parameter `kind` (valid values are `dataset`/`grid`) to make the function working for both tables and grids.

Reply:
There would a lot of if-thens, and the error/warning messages would be different between tables and grids though.

Reply:
Ah right, tables use `outfile` but grids use `outgrid`.

Reply:
> Just on the naming, should we specify that this applies to tabular outputs? E.g. use something like `validate_output_table_type`?

Renamed in becbb739b8c78d28684583bf7686e597917dd7b0.